docs/advanced/misc.rst

```diff
@@ -237,9 +237,9 @@ avoids this issue involves weak reference with a cleanup callback:
 
 .. note::
 
-    PyPy (at least version 5.9) does not garbage collect objects when the
-    interpreter exits. An alternative approach (which also works on CPython) is to use
-    the :py:mod:`atexit` module [#f7]_, for example:
+    PyPy does not garbage collect objects when the interpreter exits. An alternative
```

Comment:
We might want to re-confirm, this, though. I did not do that, yet.

Reply:
We have tests for that. Some tests explicitly call `gc.collect()` more than once because of PyPy, so it shouldn't be hard to verify.

Reply:
I'll give it a quick try, but it's actually documented: https://doc.pypy.org/en/latest/cpython_differences.html#differences-related-to-garbage-collection-strategies

> Last note: CPython tries to do a `gc.collect()` automatically when the program finishes; not PyPy. (It is possible in both CPython and PyPy to design a case where several `gc.collect()` are needed before all objects die. This makes CPython’s approach only work “most of the time” anyway.)

Should we add this to the docs, or are we fine as is?

Reply:
(I actually can't find any tests on the garbage collector passing by when the interpreter exits, though. So not sure how to apply @bstaletic's suggestion.)

Reply:
What about calling `PyGC_Collect()`? Would that work across the board?

Reply:
Good plan, but seems PyPy doesn't have `PyGC_Collect` :-(

Reply:
Other than `gc.collect()`, which isn't C, there's `PyGC_Collect()` and a bunch of "didn't mean to make it public" functions that got removed in python 3.9. On the other hand `PyGC_Collect()` was never documented.

Conclusion: CPython is a horrible mess.

Reply:
The PyPy devs, struggling to keep up with massive amounts of C API every release, are the last ones you need to convince of that statement, I'm afraid ;-)